tests/net/trickle/src/main.c

```diff
@@ -79,111 +79,74 @@ static void cb_2(struct net_trickle *trickle, bool do_suppress,
 	cb_2_called = true;
 }
 
-static bool test_trickle_create(void)
+static void test_trickle_create(void)
 {
 	int ret;
 
 	ret = net_trickle_create(&t1, T1_IMIN, T1_IMAX, T1_K);
-	if (ret) {
-		TC_ERROR("Trickle 1 create failed\n");
-		return false;
-	}
+	zassert_false(ret, "Trickle 1 create failed");
 
 	ret = net_trickle_create(&t2, T2_IMIN, T2_IMAX, T2_K);
-	if (ret) {
-		TC_ERROR("Trickle 2 create failed\n");
-		return false;
-	}
-
-	return true;
+	zassert_false(ret, "Trickle 2 create failed");
 }
 
-static bool test_trickle_start(void)
+static void test_trickle_start(void)
 {
 	int ret;
 
 	ret = net_trickle_start(&t1, cb_1, &t1);
-	if (ret) {
-		TC_ERROR("Trickle 1 start failed\n");
-		return false;
-	}
+	zassert_false(ret, "Trickle 1 start failed");
 
 	ret = net_trickle_start(&t2, cb_2, &t2);
-	if (ret) {
-		TC_ERROR("Trickle 2 start failed\n");
-		return false;
-	}
-
-	return true;
+	zassert_false(ret, "Trickle 2 start failed");
 }
 
-static bool test_trickle_stop(void)
+static void test_trickle_stop(void)
 {
-	int ret;
+	zassert_false(net_trickle_stop(&t1),
+			"Trickle 1 stop failed");
 
-	ret = net_trickle_stop(&t1);
-	if (ret) {
-		TC_ERROR("Trickle 1 stop failed\n");
-		return false;
-	}
-
-	ret = net_trickle_stop(&t2);
-	if (ret) {
-		TC_ERROR("Trickle 2 stop failed\n");
-		return false;
-	}
-
-	return true;
+	zassert_false(net_trickle_stop(&t2),
+			"Trickle 2 stop failed");
 }
 
-static bool test_trickle_1_status(void)
+static void test_trickle_1_status(void)
 {
 	if (!net_trickle_is_running(&t1)) {
-		TC_ERROR("Trickle 1 not running\n");
-		return false;
+		zassert_true(0, "Trickle 1 not running");
 	}
```

Comment:
you should get rid of the if statement and do the checking in the zassert, same below